src/java.base/share/classes/java/util/zip/ZipEntry.java

```diff
@@ -702,4 +710,20 @@ public Object clone() {
             throw new InternalError(e);
         }
     }
+
+    /**
+     * Validate that the CEN header size + name length + comment length + extra length
+     *  do not exceed 65,535 bytes per the PKWare APP.NOTE
+     *  4.4.10, 4.4.11, & 4.4.12.
+     * @param name Zip entry name
+     * @param extra Zip extra data
+     * @param comment Zip entry comment
+     * @return true if valid CEN Header size; false otherwise
+     */
+     static boolean isCENHeaderValid(String name, byte[] extra, String comment) {
```

Comment:
I think it should be spelled out better that `isCENHeaderValid` can give false positives since both `name` and `comment` may turn into more than `String::length` bytes after conversion. So while it's a reasonable sanity check to fail-fast you still need an exact check after conversion like in `ZipOutputStream::writeCEN`. Or we could consider converting `name` and `comment` to bytes eagerly and get an exact check up front?

Reply:
Sure, I can add an additional comment here if you like for future maintainers.  I don't want to consider converting name/comments earlier than needed as part of this PR as that start to creep out of the scope of the original issue being addressed, especially given this is a corner case(a corpus search of 90K+ jars only found 520 Zip entries with a header size between 500 - 1000 bytes and everything else  < 500 bytes).

Reply:
Good

Yes, converting early might have other compatibility concerns, too.